back/app/controllers/web_api/v1/admin_publications_controller.rb

```diff
@@ -9,9 +9,18 @@ def index
     publications = policy_scope(AdminPublication.includes(:parent))
     publications = publication_filterer.filter(publications, params.merge(current_user: current_user))
 
-    @publications = publications.includes(:publication, :children)
-      .order(:ordering)
-    @publications = paginate @publications
+    # A flattened ordering, such that project publications with a parent (projects in folders) are ordered
+    # first by their parent's :ordering, and then by their own :ordering (their ordering within the folder).
+    publications = publications.select(
```

Comment:
I think it could be interesting to move some of this logic in a separate method or scope or so. For example, a scope `order_global_flattened` (ideally a better name) on the Project model. Although then it becomes tricky to get AdminPublications in the same order as the projects (so a `order_leaves_flattened` scope on the AdminPublication might be easer).

Reply:
Or perhaps this should just be the default scope of admin publications?

Reply:
Good to know, yes I'm fine leaving it like this :)